snap/snapcraft.yaml

```diff
@@ -126,21 +137,83 @@ parts:
       usr/share/seabios/kvmvapic.bin: qemu/kvmvapic.bin
       usr/lib/ipxe/qemu/efi-virtio.rom: qemu/efi-virtio.rom
 
-  dnsmasq:
-    plugin: nil
-    stage-packages:
-    - dnsmasq
-
   kvm-support:
     plugin: nil
     stage-packages:
     - msr-tools
 
+  libvirt:
+    source: .
+    source-subdir: libvirt-1.3.1
+    plugin: autotools
+    build-packages:
+    - libxml2-dev
+    - libxml-libxml-perl
+    - libcurl4-gnutls-dev
+    - libncurses5-dev
+    - libreadline-dev
+    - zlib1g-dev
+    - libgcrypt20-dev
+    - libgnutls28-dev
+    - libyajl-dev
+    - libpcap0.8-dev
+    - libaudit-dev
+    - libdevmapper-dev
+    - libpciaccess-dev
+    - libnl-3-dev
+    - libnl-route-3-dev
+    - uuid-dev
+    - libnuma-dev
+    - python-all
+    - python-six
+    stage-packages:
+    - dmidecode
+    - dnsmasq
+    - libxml2
+    - libyajl2
+    - libnuma1
+    - libcurl3-gnutls
+    - libpciaccess0
+    configflags:
+    - --with-qemu
+    - --without-bhyve
+    - --without-xen
+    - --without-openvz
+    - --without-vmware
+    - --without-xenapi
+    - --without-esx
+    - --without-hyperv
+    - --without-lxc
+    - --without-vz
+    - --without-vbox
+    - --without-uml
+    - --without-sasl
+    - --without-storage-iscsi
+    - --without-storage-sheepdog
+    - --without-storage-rbd
+    - --without-storage-lvm
+    - --without-selinux
+    - --prefix=/snap/$SNAPCRAFT_PROJECT_NAME/current
+    - --localstatedir=/var/snap/$SNAPCRAFT_PROJECT_NAME/common
+    - --sysconfdir=/var/snap/$SNAPCRAFT_PROJECT_NAME/common
+    - DNSMASQ=/snap/$SNAPCRAFT_PROJECT_NAME/current/usr/sbin/dnsmasq
+    - DMIDECODE=/snap/$SNAPCRAFT_PROJECT_NAME/current/usr/sbin/dmidecode
+    prepare: |
+      apt update
+      apt source libvirt
```

Comment:
Oh, tricky. Are we sure we want "old" libvirt like that?

Reply:
I figure we want to use the version of libvirt found in Xenial since that's the target distro version for built snaps.  It would be no different than listing `libvirt` in `stage-packages` accept I had to rebuild libvirt to account for the hard coded paths.

Reply:
Well, snaps are supposed to be target agnostic, not "target xenial". Any case, my question was whether it would be more interesting to use newer libvirt? Fine, if not.

Reply:
Yeah it's heavily patched by our security team:
https://launchpad.net/ubuntu/+source/libvirt/1.3.1-1ubuntu10.19

Reply:
Yes, Ubuntu patches libvirt quite a bit.  And my thinking is that I tried to just use libvirt in stage-packages, but the only problem was hard coded paths that are set during compile time.  So I'm basically taking what would be given in stage-packages and instead building it from package source with new paths that work in the snap...and maybe a little tweaking on what actually gets built.

Reply:
OK, in that case, probably better to keep it as is.